### ISSUE TITLE
"Suggest a change on this page" button bug/change request

### ISSUE THREAD

**Issue by JavierLopezT** [apache/airflow#16654]
Coming from google to the following web: https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/s3/index.html

The button 'Suggest a change on this page' doesn't work, it leads you to https://github.com/apache/airflow/edit/main/docs/apache-airflow-providers-amazon/_api/airflow/providers/amazon/aws/hooks/s3/index.rst
<img width="866" alt="Captura de pantalla 2021-06-25 a las 14 36 56" src="https://user-images.githubusercontent.com/11339132/123425745-cc20b780-d5c2-11eb-8963-3eb10ee6db5c.png">

Also, I think the button is way too big and it makes "difficult" to read the documentation
<img width="1070" alt="Captura de pantalla 2021-06-25 a las 14 38 04" src="https://user-images.githubusercontent.com/11339132/123425890-f5414800-d5c2-11eb-97e6-725b3c909676.png">

**Comment by potiuk** [comment]
Yep. Agree. The API documentation is automatically generated and the "suggest a change" button makes no sense there. Maybe you could try to figure out how to remove it @JavierLopezT and make a PR to fix it ? @mik-laj might provide some hints.

**Comment by ShivamJhaa** [comment]
I will like to fix this bug.
Please assign me the task.
Thank you

**Comment by potiuk** [comment]
assigned!

**Comment by eladkal** [comment]
@ShivamJhaa  are you still working on this issue?

**Comment by Megha-Sai-Sree-PINNAKA** [comment]
Hello, is this issue still available? If yes, I would like to try it.

**Comment by potiuk** [comment]
Sure.

**Comment by potiuk** [comment]
Assigned you

**Comment by Megha-Sai-Sree-PINNAKA** [comment]
Hello @potiuk. You mentioned that removing the suggest a change button would be better right as the documentation is automatically generated. Do you mean removing it in  all the web pages or only from here https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/s3/index.html?

**Comment by potiuk** [comment]
All the auto-generated pages - those that have no corresponding .rst file but are generated from the python code.

**Comment by Megha-Sai-Sree-PINNAKA** [comment]
Hello @potiuk. I understood that all the pages which doesn't have .rst files but are being generated by python code (suggest a change button from these pages) has to be removed. Please correct me if my understanding is wrong.

**Comment by Megha-Sai-Sree-PINNAKA** [comment]
Hello @potiuk and @mik-laj

I understood that there are 2 ways to display the documentation on [airflow website](https://airflow.apache.org/docs/apache-airflow/stable/). 
1) Through .rst files where any changes can be directly done on this file and will be reflected on the corresponding UI Page.
2) Through .py file. In case of [amazon aws hooks](https://github.com/apache/airflow/tree/main/airflow/providers/amazon/aws/hooks) (s3)the decorators and static methods are being used for the document generation on the HTML Page. But don’t have a default .rst file to make any direct changes so that it gets displayed on the UI.

Can you please provide me any pointers to know how the [airflow site pages](https://github.com/apache/airflow-site/blob/main/sphinx_airflow_theme/sphinx_airflow_theme/suggest_change_button.html) are being invoked from [apache airflow repo](https://github.com/apache/airflow)?

**Comment by potiuk** [comment]
The .py files are generated by autoapi plugin - that's what I know. once the build is done, they are all generated in _api subfolder, so the most likely way to check it is to use the location of the source. But how exactly to do it, I am not sure - I am not a sphinx expert :)

**Comment by mik-laj** [comment]
Documentation for class is generated based on python files.

**Comment by qleroy** [comment]
The button is still way too large and makes parts of the page unreadable.
<img width="399" alt="Screenshot 2023-08-11 at 09 20 24" src="https://github.com/apache/airflow/assets/9881262/c85c7601-0b33-454c-8150-5b69766000ed">

**Comment by shivannakarthik** [comment]
@potiuk - If this is still open, I would like to pick it up.

**Comment by potiuk** [comment]
@shivannakarthik We are unassigning you from this issue as part of our updated [assignment policy](https://github.com/apache/airflow/blob/main/contributing-docs/04_how_to_contribute.rst#contribute-code-changes).

This is not meant to discourage your contribution — quite the opposite! You are still very welcome to work on this issue and submit a PR for it. Simply comment that you are working on it and open a PR when ready.

We found that formal assignments were not working well, as they often prevented others from contributing when the assignee was not actively working on the issue.

**Comment by shivannakarthik** [comment]
@potiuk Apologies, had to focus on some other priorities. Got some time to look at this. 
Could you please review? All CI checks have passed. 

I am working on another PR to reduce the size of the button itself, perhaps just having a GitHub Icon with "Edit" text, to keep it crisp and simple. Will share that PR soon

**Comment by shivannakarthik** [comment]
@potiuk - Could you please review this? A follow up PR for the size of the edit icon.
https://github.com/apache/airflow-site/pull/1572
